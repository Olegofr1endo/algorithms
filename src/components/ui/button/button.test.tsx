import React from "react";
import { Button } from "./button";
import { render, screen, fireEvent } from '@testing-library/react';
import TestRenderer from "react-test-renderer";


const callback = ()=>{
    alert("success")
}

describe("Button component tests", ()=>{
    it("with text", ()=>{
        const tree = TestRenderer.create(<Button text="SomeText"/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("without text", ()=>{
        const tree = TestRenderer.create(<Button/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it ('onload', ()=>{
        const tree = TestRenderer.create(<Button isLoader={true}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('disabled', ()=>{
        const tree = TestRenderer.create(<Button disabled={true}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("event", ()=>{
        window.alert = jest.fn();

        render(<Button text="testButton" onClick={callback}/>)

        const button = screen.getByText("testButton");

        fireEvent.click(button)

        expect(window.alert).toHaveBeenCalledWith("success")
    })
})

