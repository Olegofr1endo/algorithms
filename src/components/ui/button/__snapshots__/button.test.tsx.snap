// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component tests disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button component tests onload 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    data-cypress="loader"
    src="loader.svg"
  />
</button>
`;

exports[`Button component tests with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    SomeText
  </p>
</button>
`;

exports[`Button component tests without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
